### PR TITLE
Sudorule test enhancement

### PR DESCRIPTION
**Pull request by rjeffman**
This patch adds tests for some options that were not being tested, and enhances test behavior.

### CODE REVIEW

File: tests/sudorule/test_sudorule.yml [hunk 1]
```diff
@@ -123,6 +305,125 @@
     register: result
     failed_when: result.changed
 
+  # rafasgj
```
Comment (t-woerner) [inline tests/sudorule/test_sudorule.yml]: What is `rafasgj`?

File: tests/sudorule/test_sudorule.yml [hunk 2]
```diff
@@ -53,11 +87,87 @@
     register: result
     failed_when: result.changed
 
-  - name: Ensure sudorule is present, runAsUserCategory.
+  - name: Add user01 to the list of users sudorule execute as.
```
Comment (t-woerner) [inline tests/sudorule/test_sudorule.yml]: All adds and removes here should be ensure presence/absence of ...

File: tests/sudorule/test_sudorule.yml (hunk 1 above)
Comment (rjeffman) [reply tests/sudorule/test_sudorule.yml]: Myself. It's a note left behind. Sorry for that.

File: tests/sudorule/test_sudorule.yml (hunk 2 above)
Comment (rjeffman) [reply tests/sudorule/test_sudorule.yml]: fixed.

**Pull request merged**